src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -253,6 +256,35 @@ ORDER BY CONCAT(user.firstName, ' ', user.lastName)
             """)
     List<User> searchByNameInGroups(@Param("groupNames") Set<String> groupNames, @Param("nameOfUser") String nameOfUser);
 
+    @Query("""
+            SELECT user.id
+            FROM User user
+            LEFT JOIN user.groups userGroup
+            WHERE user.isDeleted = FALSE
+              AND :groupName = userGroup
+              AND (
+                  user.login LIKE %:loginOrName%
+                  OR CONCAT(user.firstName, ' ', user.lastName) LIKE %:loginOrName%
```

Comment:
Why is the behavior here different compared to the query for StudentParticipations:
```
p.student.firstName LIKE %:partialStudentName%
OR p.student.lastName LIKE %:partialStudentName%
```